server/routes/activities/create-an-activity/handlers/checkPay.ts

```diff
@@ -29,60 +27,12 @@ export default class CheckPayRoutes {
   }
 
   POST = async (req: Request, res: Response): Promise<void> => {
-    const { user } = res.locals
     const { pay, flat } = req.session.createJourney
 
     if (pay.length === 0 && flat.length === 0) {
       return res.validationFailed('', `Add at least one pay rate`)
     }
 
-    const minimumIncentiveLevel = await this.prisonService.getMinimumIncentiveLevel(
-      user.activeCaseLoadId,
-      user,
-      pay,
-      flat,
-    )
-    req.session.createJourney.minimumIncentiveNomisCode = minimumIncentiveLevel.levelCode
-    req.session.createJourney.minimumIncentiveLevel = minimumIncentiveLevel.levelName
-
-    if (req.params.mode === 'edit') {
-      const { activityId } = req.session.createJourney
-      const prisonCode = user.activeCaseLoadId
-      const activity = {
-        paid: true,
-        minimumIncentiveLevel: req.session.createJourney.minimumIncentiveLevel,
-        minimumIncentiveNomisCode: req.session.createJourney.minimumIncentiveNomisCode,
-        pay: req.session.createJourney.pay.map(p => ({
-          incentiveNomisCode: p.incentiveNomisCode,
-          incentiveLevel: p.incentiveLevel,
-          payBandId: p.prisonPayBand.id,
-          rate: p.rate,
-        })),
-      } as ActivityUpdateRequest
-
-      if (req.session.createJourney.flat && req.session.createJourney.flat.length > 0) {
-        const incentiveLevels = await this.prisonService.getIncentiveLevels(user.activeCaseLoadId, user)
-
-        req.session.createJourney.flat.forEach(flatRate => {
-          incentiveLevels.forEach(iep =>
-            activity.pay.push({
-              incentiveNomisCode: iep.levelCode,
-              incentiveLevel: iep.levelName,
-              payBandId: flatRate.prisonPayBand.id,
-              rate: flatRate.rate,
-            }),
-          )
-        })
-      }
-
-      await this.activitiesService.updateActivity(prisonCode, activityId, activity)
-      const successMessage = `You've updated the pay for ${req.session.createJourney.name}`
-
-      const returnTo = `/activities/view/${req.session.createJourney.activityId}`
-      req.session.returnTo = returnTo
-      return res.redirectOrReturnWithSuccess(returnTo, 'Activity updated', successMessage)
-    }
```

Comment:
I think this code was left behind from when we used to treat the review pay page as check answers page for editing pay.

Now that the add, update and remove pay handlers all call the `updateActivity()` immediately after a change has been made this update logic is no longer used.